File: backend/spring-boot/src/main/java/com/runhwani/runmate/service/UserService.java
package com.runhwani.runmate.service;

import com.runhwani.runmate.dto.response.user.UserProfileResponse;

/**
 * 사용자 관련 서비스 인터페이스
 */
public interface UserService {
    
    /**
     * 현재 로그인한 사용자의 프로필 정보를 조회합니다.
     * @return 사용자 프로필 정보
     */
    UserProfileResponse getCurrentUserProfile();
} 

File: RunMate/data/src/main/java/com/D107/runmate/data/remote/api/UserService.kt
package com.D107.runmate.data.remote.api

import com.D107.runmate.data.remote.common.ApiResponse
import com.D107.runmate.data.remote.common.ServerResponse
import com.D107.runmate.data.remote.request.user.LoginRequest
import com.D107.runmate.data.remote.response.user.LoginResponse
import com.D107.runmate.data.remote.response.user.SignupResponse
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.Part

interface UserService {
    @POST("auth/login")
    suspend fun login(
        @Body loginRequest: LoginRequest
    ): Response<ServerResponse<LoginResponse>>

    @Multipart
    @POST("auth/signup")
    suspend fun signup(
        @Part("data") signupRequest: RequestBody,
        @Part profileImage: MultipartBody.Part?
    ): Response<ServerResponse<SignupResponse>>
}